runtime/params.ts

```diff
@@ -140,10 +142,21 @@ export class SkyAppRuntimeConfigParams {
   /**
    * Returns the params object
    * @name getAll
+   * @param {boolean} excludeDefaults Exclude params that have default values
    * @returns {Object}
    */
-  public getAll(): Object {
-    return this.params;
+  public getAll(excludeDefaults?: boolean): Object {
+    const filter = (key: string) => excludeDefaults ?
+      this.params[key] !== this.defaultParamValues[key] : true;
+
+    const reduce = (params: { [key: string]: string }, key: string) => {
+      params[key] = this.params[key];
+      return params;
+    };
+
+    return this.getAllKeys()
```

Comment:
My suggestion here would be to write a simple loop to build up the object.  It'll cut down on both the amount of code written as well as the number of loops executed (1 vs. 2) and be more legible IMO.

```
const filteredParams: { [key: string]: string} = {};

this.getAllKeys().forEach(key => {
  if (!excludeDefaults || this.params[key] !== this.defaultParamValues[key]) {
    filteredParams[key] = this.params[key];
  }
});

return filteredParams;
```